pkg/webhook/resources/virtualmachineimage/validator.go

```diff
@@ -70,8 +70,12 @@ func (v *virtualMachineImageValidator) Create(request *types.Request, newObj run
 
 	if newImage.Spec.SourceType == v1beta1.VirtualMachineImageSourceTypeDownload && newImage.Spec.URL == "" {
 		return werror.NewInvalidError(`url is required when image source type is "download"`, "spec.url")
-	} else if newImage.Spec.SourceType == v1beta1.VirtualMachineImageSourceTypeUpload && newImage.Spec.URL != "" {
-		return werror.NewInvalidError(`url should be empty when image source type is "upload"`, "spec.url")
+	} else if newImage.Spec.SourceType != v1beta1.VirtualMachineImageSourceTypeDownload && newImage.Spec.URL != "" {
+		return werror.NewInvalidError(`url should be empty when image source type is not "download"`, "spec.url")
+	} else if newImage.Spec.SourceType == v1beta1.VirtualMachineImageSourceTypeExportVolume {
```

Comment:
Can we add a permission check to ensure the user has related PVC permission when it references the PVC in the image object?

Reply:
@gitlawr updated.